src/client.c

```diff
@@ -427,23 +433,29 @@ bool handle_auth_query_response(PgSocket *client, PktHdr *pkt) {
 			 * NULL - set an md5 password with an impossible value,
 			 * so that nothing will ever match
 			 */
-			password = "md5";
+			passwd = "md5";
 			length = 3;
 		} else {
-			if (!mbuf_get_chars(&pkt->data, length, &password)) {
+			if (!mbuf_get_chars(&pkt->data, length, &passwd)) {
 				disconnect_server(server, false, "bad packet");
 				return false;
 			}
 		}
-		if (sizeof(user.passwd)  - 1 < length)
-			length = sizeof(user.passwd) - 1;
-		memcpy(user.passwd, password, length);
+		if (sizeof(real_passwd) - 1 < length)
+			length = sizeof(real_passwd) - 1;
+		memcpy(real_passwd, passwd, length);
 
-		client->login_user = add_db_user(client->db, user.name, user.passwd);
+		client->login_user = add_user(real_user, "");
```

Comment:
We can leave `PgUser*->passwd` empty here because we've just received the user's password from Postgres `pg_shadow`. This means that the user's password must not be in `auth_file`.